amxmodx/amxmodx.cpp

```diff
@@ -23,15 +23,30 @@
 extern CFlagManager FlagMan;
 ke::Vector<CAdminData *> DynamicAdmins;
 
+const char *g_sInaccessibleXVars[] =
+{
+	"MaxClients",
+	"MapName",
+	"PluginName",
+	"PluginVersion",
+	"PluginAuthor",
+	"PluginURL",
+	"NULL_STRING",
+	"NULL_VECTOR"
+};
+
 static cell AMX_NATIVE_CALL get_xvar_id(AMX *amx, cell *params)
 {
 	int len;
 	char* sName = get_amxstring(amx, params[1], 0, len);
 	cell ptr;
 
-	if (!strcmp(sName, "MaxClients") || !strcmp(sName, "MapName") || !strcmp(sName, "NULL_STRING") || !strcmp(sName, "NULL_VECTOR"))
+	for (auto name : g_sInaccessibleXVars)
 	{
-		return -1;
+		if (likely(!strcmp(sName, name)))
```

Comment:
I'm not totally sure it's going to be useful in this context. Also, this is _unlikely_ that forbidden xvars are going to be used here.

Reply:
@Arkshine and what's stopping me from creating a xvar with a name like PluginName?

Reply:
I was talking about `likely` only, not the actual check for forbidden identifiers.

Reply:
@WPMGPRoSToTeMa any thoughts about _likely_/_unlikely_ here?

Reply:
@Arkshine this is recommendation from @In-line (_likely_). Okay, I'll clean this up. ))

Reply:
@Arkshine yeah, this is absolutely `unlikely`. But I think `unlikely` should be automatically deduced by the compiler in case of early return.